bin/start.sh

```
@@ -677,25 +723,102 @@ ensure_prisma_client_fresh() {
+    else
+      ((up_to_date++)) || true
+      # Ensure already-built plugins are CDN-visible
+      _ensure_cdn_copy "$pname"
+    fi
   done
   if [ ${#to_build[@]} -gt 0 ]; then
-    log_warn "Plugins need building: ${to_build[*]}"
+    log_warn "Plugins need building: ${to_build[*]} ($up_to_date already up to date)"
     for p in "${to_build[@]}"; do
       [ -d "$ROOT_DIR/plugins/$p/frontend" ] || continue
       log_info "Building $p..."; cd "$ROOT_DIR/plugins/$p/frontend"
```

Shellcheck (0.11.0)

[warning] Use 'cd ... || exit' or 'cd ... || return' in case cd fails.

(SC2164)